### ISSUE TITLE
reduce vllm decode graphs

### ISSUE THREAD

**Issue by AleksKnezevic** [tenstorrent/tt-xla#4468]
Currently there are 5 graphs for each vllm decode step:

1. prepare inputs
2. main model
3. select hidden states
4. compute logits
5. sample

This should be reduced to 1/2 graphs.
